src/libCli/ConnectionManager.hpp

```diff
@@ -0,0 +1,163 @@
+            std::shared_ptr<grpc::ProtoReflectionDescriptorDatabase> getDescDb(std::string f_serverAddress, std::string f_serverPort)
+            {
+                if(f_serverPort == "")
+                {
+                    f_serverPort = "50051";
+                }
+                std::string serverAddress = f_serverAddress + ":" + f_serverPort;
+
+                if(!findDescDbByAddress(serverAddress))
+                {
+                    if(connections[serverAddress].channel)
+                    {
+                        connections[serverAddress].descDb = std::make_shared<grpc::ProtoReflectionDescriptorDatabase>(connections[serverAddress].channel);
+                        connections[serverAddress].descPool = std::make_shared<grpc::protobuf::DescriptorPool>(connections[serverAddress].descDb.get());
+                    }
+                    else
+                    {
+                        registerConnection(serverAddress);
+                    }
+                }
+                return connections[serverAddress].descDb;
+            }
+
+            std::shared_ptr<grpc::protobuf::DescriptorPool> getDescPool(std::string f_serverAddress, std::string f_serverPort)
+            {
+                if(f_serverPort == "")
+                {
+                    f_serverPort = "50051";
+                }
+                std::string serverAddress = f_serverAddress + ":" + f_serverPort;
+                if(!findDescPoolByAddress(serverAddress))
+                {
+                    if(connections[serverAddress].channel)
+                    {
+                        connections[serverAddress].descDb = std::make_shared<grpc::ProtoReflectionDescriptorDatabase>(connections[serverAddress].channel);
+                        connections[serverAddress].descPool = std::make_shared<grpc::protobuf::DescriptorPool>(connections[serverAddress].descDb.get());
+                    }
+                    else
+                    {
+                        registerConnection(serverAddress);
+                    }
+                }
+                return connections[serverAddress].descPool;
+            }
+        private:
+            // connection cache for resuing the channel, corresponding descriptor Database and DatabasePool
+            std::unordered_map<std::string, ConnList> connections;
+            void registerConnection(std::string f_serverAddress)
+            {
+                ConnList connection;
+                connection.channel = grpc::CreateChannel(f_serverAddress, grpc::InsecureChannelCredentials());
+                connection.descDb = std::make_shared<grpc::ProtoReflectionDescriptorDatabase>(connection.channel);
+                connection.descPool = std::make_shared<grpc::protobuf::DescriptorPool>(connection.descDb.get());
+
+                if(connections.find(f_serverAddress) == connections.end())
+                {
+                    connections.insert(std::make_pair(f_serverAddress, connection));
```

Comment:
why not `connections[f_serverAddress] = connection;` ?
This will automatically insert if not exists